File: server/src/main/java/org/hkurh/doky/security/DokyAuthority.java
package org.hkurh.doky.security;

import org.springframework.security.core.GrantedAuthority;

public enum DokyAuthority implements GrantedAuthority {
    ROLE_USER(Role.ROLE_USER);

    private final String authority;

    DokyAuthority(String authority) {
        this.authority = authority;
    }

    @Override
    public String getAuthority() {
        return authority;
    }

    public static class Role {
        public static final String ROLE_USER = "ROLE_USER";
    }
}

File: server/src/main/java/org/hkurh/doky/security/DokyAuthority.kt
package org.hkurh.doky.security

import org.springframework.security.core.GrantedAuthority

enum class DokyAuthority(private val authority: String) : GrantedAuthority {
    ROLE_USER(Role.ROLE_USER);

    override fun getAuthority(): String {
        return authority
    }

    object Role {
        const val ROLE_USER = "ROLE_USER"
    }
}
